test/Microsoft.AspNetCore.ResponseCaching.Tests/ResponseCacheMiddlewareTests.cs

```diff
@@ -28,24 +32,45 @@ public class ResponseCacheMiddlewareTests
 
             Assert.True(await middleware.TryServeFromCacheAsync(context));
             Assert.Equal(StatusCodes.Status504GatewayTimeout, context.HttpContext.Response.StatusCode);
+            Assert.Equal(2, sink.Writes.Count);
+            TestUtils.AssertLoggedMessageContains(
+                sink.Writes[0],
+                LogLevel.Debug,
+                "Retrieving cache entry for base key");
```

Comment:
Actually it's probably better to use eventIDs instead of string matches. More specific and less verbose